doc/design/garbage-collection-1.md

```diff
@@ -0,0 +1,52 @@
+# Garbage collection
+There are several kinds of data that exist without being included in a dandiset, meaning they are simply taking up space in the system.
+These data should be cleaned up automatically after some time.
+
+# Data types
+## Uploads
+As part of the multipart upload process, an `Upload` database record is created to track the status of the upload.
+`Upload`s can be orphaned when a client aborts the upload process before creating an `AssetBlob` from the `Upload`.
+
+`Uploads` could be abandoned prior to any uploads, partially uploaded, or after upload is complete.
+The preferred way to deal with partial uploads in S3 is to set up a [bucket lifecycle rule](https://aws.amazon.com/blogs/aws/s3-lifecycle-management-update-support-for-multipart-uploads-and-delete-markers/).
+If the object is already uploaded, we can simply delete it from S3.
+In all cases, we need to delete the database record.
+
+`Upload`s should be deleted whenever the presigned URLs expire, which is currently set in `multipary.py#DandiMultipartMixin` to 7 days.
+
+## AssetBlobs
+After an upload is complete, it is converted into an `AssetBlob`, which is then registered with an `Asset`.
+If a user skips this registration step, the file will be available in S3, but will not appear in any dandisets.
```

Comment:
What's the usual step for registering an `AssetBlob`? Why do we allow this step to not be taken?